client/state/ui/editor/contact-form/test/reducer.js

```diff
@@ -171,4 +173,122 @@ describe( "editor's contact form state reducer", () => {
 			assert.notStrictEqual( state.fields, CONTACT_FORM_DEFAULT.fields, 'the fields on the returned state and the default contact form are strictly equal.' );
 		} );
 	} );
+
+	describe( 'update field', () => {
+		it( 'should update a field by index', () => {
+			const contactForm = {
+				fields: [
+					{ label: 'Name' },
+					{ label: 'Email' },
+					{ label: 'Website' },
+					{ label: 'Comment' }
+				]
+			};
+
+			const state = reducer( contactForm, {
+				type: EDITOR_CONTACT_FORM_UPDATE_FIELD,
+				index: 2,
+				field: { label: 'Web Address', type: 'url', required: true }
```

Comment:
>Optional: use `deepFreeze` to detect mutations on contactForm

Required in my opinion :smile: At minimum, very strongly recommended. It's caught me a few times.